src/psyclone/dynamo0p3.py

```diff
@@ -2644,6 +2932,27 @@ def _setup(self, ktype, module_name, args, parent):
             self._qr_name = self._name_space_manager.create_name(
                 root_name=qr_arg.varName, context="AlgArgs",
                 label=self._qr_text)
+            # dynamo 0.3 api kernels require quadrature rule arguments to be
+            # passed in if one or more basis functions are used by the kernel
+            # and gh_shape == "gh_quadrature_***".
+            self._qr_args = {"nh": "nqp_h",
+                             "nv": "nqp_v",
+                             "h": "wh",
+                             "v": "wv"}
+            # If we're not a kernel stub then we will have a name for the qr
+            # argument. We append this to the names of the qr-related
+            # variables.
+            if qr_arg.varName:
+                for arg in self._qr_args:
+                    self._qr_args[arg] += "_" + self._qr_name
+        elif self._eval_shape == "gh_evaluator":
+            # Kernel has an evaluator. The FS of the updated argument tells
+            # us upon which nodal points the evaluator will be required
+            arg = self.updated_arg
+            if arg.is_operator:
```

Comment:
As mentioned earlier, this if test recurs and could be replaced by an internal method.